services/galley/src/Galley/API/Teams.hs

```diff
@@ -446,22 +459,43 @@ getTeamMembersCSVH (zusr ::: tid ::: _) = do
       let userMap :: M.Map UserId Handle.Handle
           userMap = M.fromList . catMaybes $ extract <$> userList
             where
-              extract u = (U.userId u,) <$> (U.userHandle u)
+              extract u = (U.userId u,) <$> U.userHandle u
 
-      pure $ (`M.lookup` userMap)
+      pure (`M.lookup` userMap)
 
     userToIdPIssuer :: U.User -> Maybe HttpsUrl
     userToIdPIssuer usr = case (U.userIdentity >=> U.ssoIdentity) usr of
       Just (U.UserSSOId issuer _) -> fromByteString' $ cs issuer
       Just _ -> Nothing
       Nothing -> Nothing
 
-    pairMembersUsers :: [TeamMember] -> [User] -> [(TeamMember, User)]
-    pairMembersUsers members users = do
-      let usersM = M.fromList (users <&> \user -> (U.userId user, user))
+    joinUserInfos :: [TeamMember] -> [User] -> [Maybe RichInfo] -> [(TeamMember, Maybe RichInfo, User)]
```

Comment:
I don't think you can rely on the order in which the `RichInfo` values are returned from the DB either.  You need to pair them with their `uid`s and make that another map, like `userMap`.